token.go

```diff
@@ -37,12 +37,18 @@ type PacketAndToken struct {
 type Token interface {
 	// Wait will wait indefinitely for the Token to complete, ie the Publish
 	// to be sent and confirmed receipt from the broker.
+	//
```

Comment:
I'm not sure if we want to add this to the interface definition because this formalises the behaviour. Currently it's undocumented which leaves some room to change it (which probably should probably happen - for the reconnect anyway - its just a matter of someone putting the time in to make the change).